### DISCUSSION TITLE
Decrypting Environment Vars in REST API

### DISCUSSION THREAD

**Discussion by jfrosch** [vercel/community#687] [Help] [answered]
When fetching environment variables via REST API, per the docs, I set a decrypt query parameter to true, but the response still has an encrypted value. I must be missing something obvious.

The URL looks like this:
```
https://api.vercel.com/v9/projects/prj_.../env?decrypt=true&teamId=team_...
```
Each environment variable object in the array of env vars fetched comes back like this:
```
{
    type: 'encrypted',
    key: 'SAMPLE_ENV',
    value: 'mGH8pOy3KxMacJLBa3a5M91dxAeaobNB4FlMC0GLzmM=',
    target: [ 'production', 'preview', 'development' ],
    configurationId: null,
    createdAt: 1655759371216,
    updatedAt: 1655847063509,
    createdBy: 'UZOKns6HiOOW7DKtjRKlfqmR',
    updatedBy: 'UZOKns6HiOOW7DKtjRKlfqmR',
    id: '1U2Nkt43nrp7XpUe',
    decrypted: false
  }
```
The API docs I referenced were [here](https://vercel.com/docs/rest-api#endpoints/projects/retrieve-the-environment-variables-of-a-project-by-id-or-name).

Note: This is a different endpoint than the one to fetch an individual env var that is decrypted.

Thanks in advance!

**Comment by swarnava** [comment]
Hi,

You need to pass `env/id` to get decrypted value:

<img width="855" alt="image" src="https://user-images.githubusercontent.com/1162991/176871107-4dc21671-7492-4762-9081-94b61c40346d.png">

Ref: https://vercel.com/docs/rest-api#endpoints/projects/retrieve-the-decrypted-value-of-an-environment-variable-of-a-project-by-id

**Reply by jfrosch** [reply]
Thanks for that info. Unfortunately, my question is for a different endpoint, the one to fetch **all** environment variables, not just a single one by id. Specifically, this [endpoint](https://vercel.com/docs/rest-api#endpoints/projects/retrieve-the-environment-variables-of-a-project-by-id-or-name).

Do I need to query the /env endpoint, get the array of all env var objects for their ids, then loop through the array fetching each by id to get the decrypted values?

**Comment by ayasa520** [comment, marked as answer]
Replacing `v9` with `v8` works fine. It looks like this API is deprecated.
![image](https://user-images.githubusercontent.com/55699692/179240103-1e1530e3-e01d-473f-8a81-85dae49af7c3.png)
